### PR TITLE
Prepare for v0.15.0 release

### PR DESCRIPTION
**Pull request by asauber**
Prepare for v0.15.0 release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ binary releases.
 
 | Release                                                              | Maintained | Supported Cilium Versions |
 |----------------------------------------------------------------------|------------|---------------------------|
-| [v0.14.8](https://github.com/cilium/cilium-cli/releases/tag/v0.14.8) | Yes        | Cilium 1.11 and newer     |
+| [v0.15.0](https://github.com/cilium/cilium-cli/releases/tag/v0.15.0) | Yes        | Cilium 1.11 and newer     |
```
Comment (michi-covalent) [inline README.md:46]: let's wait and see what @tklauser says, but i kind of want to do:

```suggestion
| [v0.15.0](https://github.com/cilium/cilium-cli/releases/tag/v0.15.0) | Yes        | Cilium 1.14 and newer     |
| [v0.14.8](https://github.com/cilium/cilium-cli/releases/tag/v0.14.8) | Yes        | Cilium 1.11, 1.12, and 1.13     |
```

for example, i don't know if Helm mode `clustermesh` commands work with cilium v1.11 👀

Comment (asauber) [reply README.md:46]: Ah, absolutely, let me add this

**Review by michi-covalent — APPROVED** [review]
🥰

File: README.md (hunk 1 above)
Comment (tklauser) [reply README.md]: I guess we could even recommend v0.15.0 as the default release for Cilium 1.13 and newer given we've tested Helm mode in cilium-cli CI with Cilium 1.13. But probably on the safer side to say Cilium 1.14.

**Pull request merged**